### ISSUE TITLE
The bindings' performance scales differently with increased order

### ISSUE THREAD

**Issue by dannys4** [MeasureTransport/MParT#184]
@rubiop noted that, as you increase the order of a map, the bindings lose performance relative to the C++ implementation. We don't know why, but it scales with the order.

**Comment by rubiop** [comment]
I made some new detail test. I created a map of order 5 in dimension 2. Then I compare:
1. Time to run `.Evaluate(pts)` in matlab
2. Time to run `->Evaluate(pts)` in C++ when `.Evaluate(pts)` in matlab is called
3. Time to run `->Evaluate(pts)` in C++ in a pure .cpp test script 

I have (mean over 50 iterations):

1. `1.3350 s`
2. `1.3192 s`
3. `0.7326 s`

When I do the same thing with `LogDeterminant`:

1. `0.0642 s`
2. `0.0631 s`
3. `0.0578 s`

I think that means that it's definitely not an issue of array conversions since `1.` and `2.`  are not significantly different. Hence, I still don't understand why `2.` and `3.` are so different for `Evaluate` and not for `LogDeterminant` and not when `order=1`.

**Comment by mparno** [comment]
I suspect this might be caused by a difference in compiler flags or compiler performance between the bindings libraries and the c++ executable.   The `EvaluateImpl` function in the `MonotoneComponent` class is defined in the header, which means that it will be compiled separately in each library or executable that uses it (e.g., `pympart.cpython-310-darwin.so` or your c++ executable).     We might need to play around with options (e.g., `-O2` instead of `-O3`) to see if we can reduce this gap.

**Comment by dannys4** [comment]
I don't think this was updated since the last meeting, but for some reason it scales oddly with the order, and this could have something to do with it. We noted something like

| order | python/C++ ratio |
| ------ | ------------------- |
| 1 | 0.8 |
| 2 | 1.05 |
| 3 | 1.2 |
| ... | ... |

and it seemed like it asymptotically approached a ratio like ~1.8 or something? I don't remember the exact number. But we suspected that the quadrature could also be providing a difference, since that seems like the main differentiator between lower and higher orders (where we don't really do much quadrature at all if we're only an order 1 map). Do you think this has anything to do with it? If I'm not mistaken, log determinant is also defined in those headers and it seems to perform fine with scaling in order.
